### PR TITLE
Automated cherry pick of #117194: Revert "Optimization on running prePreEnqueuePlugins

### PR DESCRIPTION
**Pull request by alculquicondor**
Cherry pick of #117194 on release-1.27.

#117194: Revert "Optimization on running prePreEnqueuePlugins

For details on the cherry pick process, see the [cherry pick requests](https://git.k8s.io/community/contributors/devel/sig-release/cherry-picks.md) page.

```release-note
Fix a 1.27 regression that custom PreEnqueue plugins aren't executed for Pods proceeding to activeQ through backoffQ.
```

### CODE REVIEW

**Review by xmudrii — APPROVED** [review]
/approve

File: CHANGELOG/CHANGELOG-1.27.md
Comment (xmudrii) [inline CHANGELOG/CHANGELOG-1.27.md:210]: I think this should mention v1.27.0 and v1.27.1 if I'm not mistaken.

Comment (alculquicondor) [reply CHANGELOG/CHANGELOG-1.27.md:210]: fixed

**Review by xmudrii — APPROVED** [review]
Only changelog changed:
/lgtm

**Comment by xmudrii** [comment]
/hold
Please unhold only after #117194 is merged and tests are passing.

**Pull request merged**
